scripts/carousel.js

```diff
@@ -0,0 +1,85 @@
+function setup(carousel) {
+    const items = carousel.querySelector(".carousel--items");
+    
+    items.setAttribute("alt", 0);
+    let nextBtn = carousel.querySelector(".carousel--next");
+    let prevBtn = carousel.querySelector(".carousel--prev");
+
+    setupDots(carousel, items.querySelectorAll(".carousel--item").length);
+    let dots = carousel.querySelector(".carousel--dots");
+    updateItems(items, dots);
+
+    nextBtn.addEventListener('click', next);
+    prevBtn.addEventListener('click', prev);
+}
+
+function updateItems(items, dots) {
+    let index = parseInt(items.getAttribute("alt"), 10);
+    items = items.querySelectorAll(".carousel--item");
+    for (let i = 0; i < items.length; i++) {
+        if (i != index) {
+            items[i].classList.add("carousel--item-hide");
```

Comment:
let's use the SMACSS style for these: "carousel--item-is-hidden" ("state" classes are always "is-something")